src/Entity/Building.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Entity;
+
+class Building
+{
+    private int $id;
+
+    public \DateTime $starTime;
+
+    public \DateTime $endTime;
+
+    public function getId(): int
+    {
+
+        return $this->id;
+    }
+
+    public function getStartTime() : \DateTime
+    {
+
+        return $this->startTime;
+    }
+
+    public function setStartTime(\DateTime $startTime): \DateTime
+    {
+
+        $this->startTime = $startTime;
+
+        return $this->startTime;
+    }
+
+    public function getEndTime() : \DateTime
+    {
+
+        return $this->endTime;
+    }
+
+    public function setEndTime(\DateTime $endTime): \DateTime
```

Comment:
You should hint the return type as 'self' and return $this. This technique is used to chain methods when calling them.
Here is a useful link for understanding how it works : https://riptutorial.com/php/example/2190/method-chaining-in-php

Reply:
This applies to setters